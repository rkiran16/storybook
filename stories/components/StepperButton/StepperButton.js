import { html } from "lit-html";

export const StepperButton = ({ disabled = false, iconName }) => {

  return html`
    <button
      type="button"
      ?disabled=${disabled}
      class="btn btn-icon btn-stepper"
    >
      <span class="material-icons">${iconName}</span>
    </button>
  `;
};
