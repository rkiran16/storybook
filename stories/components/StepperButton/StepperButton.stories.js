import { StepperButton } from './StepperButton';

export default {
  title: 'Components/Buttons/StepperButton',
  argTypes: {
    iconName: {
      name: "Stepper Increment/Decrement",
      control: { type: 'select'},
      description: "Select Icon Name",
      defaultValue: "add",
      options: ['add', 'remove'],
    },
    disabled: {
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    }
  },
};

const Template = (args) => StepperButton(args);


export const Standard = Template.bind({});
Standard.parameters = {
  layout: 'centered',
};

Standard.args = {
   disabled: false
}
