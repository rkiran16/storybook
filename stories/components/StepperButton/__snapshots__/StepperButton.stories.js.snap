// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Buttons/StepperButton Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <button
      type=\\"button\\"
      ?disabled=",
    "
      class=\\"btn btn-icon btn-stepper\\"
    >
      <span class=\\"material-icons\\">",
    "</span>
    </button>
  ",
  ],
  "values": Array [
    false,
    "add",
  ],
}
`;
